File: app/src/main/java/com/magenta/rx/java/http/DictionaryClient.java
package com.magenta.rx.java.http;

import com.magenta.rx.java.model.record.DictionaryAnswer;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

public interface DictionaryClient {

    @GET(value = "/api/v1/dicservice.json/lookup")
    Observable<DictionaryAnswer> dictionary(@Query("key") String key, @Query("text") String text, @Query("lang") String lang);
}

File: app/src/main/kotlin/com/magenta/rx/kotlin/http/DictionaryClient.kt
package com.magenta.rx.kotlin.http

import com.magenta.rx.kotlin.record.DictionaryAnswer
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

interface DictionaryClient {

    @GET("/api/v1/dicservice.json/lookup") fun dictionary(@Query("key") key: String, @Query("text") text: String, @Query("lang") lang: String): Observable<DictionaryAnswer>
}